Test.Activities.Automation.ActivityLib/Models/Activity/Services/SyncActivityService.cs

```diff
@@ -0,0 +1,246 @@
+
+                var spMentorsList = web.Lists.TryGetList(Constants.Lists.Mentors);
+                if (spMentorsList == null) throw new Exception("Getting SP mentor list failed");
+
+                var spRootMentorsList = web.Lists.TryGetList(Constants.Lists.RootMentors);
+                if (spRootMentorsList == null) throw new Exception("Getting SP root mentor list failed");
+
+                var members = new List<Member>();
+
+                foreach (var item in spMentorsList.GetItems().Cast<SPListItem>())
+                {
+                    var mentor = SPHelper.GetUserValue(item, Constants.Activity.Employee);
+
+                    var paths = SPHelper.GetMultiChoiceValue(item, Constants.Activity.Paths);
+
+                    members.Add(new Member
+                    {
+                        MentorId = mentor.User.ID,
+                        MentorLookupId = item.ID,
+                        MentorEmail = mentor.User.Email,
+                        Paths = new List<string>(paths)
+                    });
+                }
+
+                foreach (var item in spRootMentorsList.GetItems().Cast<SPListItem>())
+                {
+                    var rootMentor = SPHelper.GetUserValue(item, Constants.Activity.Employee);
+
+                    var paths = SPHelper.GetMultiChoiceValue(item, Constants.Activity.Paths);
+
+                    var member = members.FirstOrDefault(x => x.MentorId == rootMentor.User.ID);
+
+                    if (member != null)
+                    {
+                        member.RootMentorId = rootMentor.User.ID;
+                        member.RootMentorEmail = rootMentor.User.Email;
+                        member.RootMentorLookupId = item.ID;
+                    }
+                    else
+                    {
+                        members.Add(new Member
+                        {
+                            RootMentorId = rootMentor.User.ID,
+                            RootMentorLookupId = item.ID,
+                            RootMentorEmail = rootMentor.User.Email,
+                            Paths = new List<string>(paths)
+                        });
+                    }
+                }
+
+                return members;
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Getting existing members from SP failed. {e.Message}");
+            }
+        }
+
+        public IEnumerable<SpActivity> GetSpActivities(SPWeb web, DateTime minDate, DateTime maxDate)
+        {
+            try
+            {
+                _logger?.LogInformation("Getting existing activities from SP");
+
+                var spListActivities = web.Lists.TryGetList(Constants.Lists.Activities);
+                if (spListActivities == null) throw new Exception("Getting SP activity list failed");
+
+                var spListMentors = web.Lists.TryGetList(Constants.Lists.Mentors);
```

Comment:
You get all members twice: here and in `GetMembers` method. Let's add one parameter `List<Member> members` to this method